src/extractor.cpp

```diff
@@ -102,20 +152,37 @@ void Extractor::way(const osmium::Way &way)
     bool forward = (!oneway || std::strcmp(oneway, "yes") == 0);
     bool reverse = (!oneway || std::strcmp(oneway, "-1") == 0);
 
-    // Check to see if it's an interesting type of way.  We're only
-    // interested in roads at the moment.
-    const char *highway = way.tags().get_value_by_key("highway");
-    bool usable =
-        highway &&
-        (std::strcmp(highway, "motorway") == 0 || std::strcmp(highway, "motorway_link") == 0 ||
-         std::strcmp(highway, "trunk") == 0 || std::strcmp(highway, "trunk_link") == 0 ||
-         std::strcmp(highway, "primary") == 0 || std::strcmp(highway, "primary_link") == 0 ||
-         std::strcmp(highway, "secondary") == 0 || std::strcmp(highway, "secondary_link") == 0 ||
-         std::strcmp(highway, "tertiary") == 0 || std::strcmp(highway, "tertiary_link") == 0 ||
-         std::strcmp(highway, "residential") == 0 || std::strcmp(highway, "living_street") == 0 ||
-         std::strcmp(highway, "unclassified") == 0 || std::strcmp(highway, "service") == 0 ||
-         std::strcmp(highway, "ferry") == 0 || std::strcmp(highway, "movable") == 0 ||
-         std::strcmp(highway, "shuttle_train") == 0 || std::strcmp(highway, "default") == 0);
+    // Check if the way contains tags we are interested in
+    // TODO, functionify
+    bool usable = false;
+    if (tags_filter.empty())
+    {
+        const char *highway = way.tags().get_value_by_key("highway");
+        usable =
+            highway &&
+            (std::strcmp(highway, "motorway") == 0 || std::strcmp(highway, "motorway_link") == 0 ||
+             std::strcmp(highway, "trunk") == 0 || std::strcmp(highway, "trunk_link") == 0 ||
+             std::strcmp(highway, "primary") == 0 || std::strcmp(highway, "primary_link") == 0 ||
+             std::strcmp(highway, "secondary") == 0 ||
+             std::strcmp(highway, "secondary_link") == 0 || std::strcmp(highway, "tertiary") == 0 ||
+             std::strcmp(highway, "tertiary_link") == 0 ||
+             std::strcmp(highway, "residential") == 0 ||
+             std::strcmp(highway, "living_street") == 0 ||
+             std::strcmp(highway, "unclassified") == 0 || std::strcmp(highway, "service") == 0 ||
+             std::strcmp(highway, "ferry") == 0 || std::strcmp(highway, "movable") == 0 ||
+             std::strcmp(highway, "shuttle_train") == 0 || std::strcmp(highway, "default") == 0);
```

Comment:
What about creating a function taking a vector of tags and checking all of them?

Or better put all tags here into a hashset and do constant time `hset.count(highway) > 1` checks